src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/VotingWidget/useVotingWidgetUpdate.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
+import { VoterRecord } from '~gql';
+import { useAppContext } from '~hooks';
+
+export const useVotingWidgetUpdate = (
+  voterRecord: VoterRecord[],
+  stopPollingAction: () => void,
+) => {
+  const { user } = useAppContext();
+
+  const currentVotingRecord = voterRecord.find(
+    ({ address }) => address === user?.walletAddress,
+  );
+
+  const [prevRecord, setPrevRecord] = useState(currentVotingRecord);
+  const [hasUserVoted, setHasUserVoted] = useState(!!currentVotingRecord);
+
+  if (currentVotingRecord && !hasUserVoted) {
+    setHasUserVoted(true);
+  }
+
+  // if user's vote count increased, db has been updated, stop polling
+  if (currentVotingRecord?.voteCount !== prevRecord?.voteCount) {
```

Comment:
There are two cases here. One where the user votes and that changes the motion state to reveal because the reputation threshold has been reached. In this case, the voting widget will be dismounted and the effect below will cancel polling.

The other is that the user votes and it doesn't change the motion state. In this case, the component won't be dismounted, so to cancel polling, we check to see if the user's vote count has increased.